src/api/procedures/inviteExternalAgent.ts

```diff
@@ -88,6 +90,14 @@ export async function prepareInviteExternalAgent(
 
   const { ticker, target, permissions, expiry } = args;
 
+  const issuer = await context.getCurrentIdentity();
+  let targetIdentity: Identity;
+  if (typeof target === 'string') {
+    targetIdentity = new Identity({ did: target }, context);
+  } else {
+    targetIdentity = target;
+  }
```

Comment:
We should probably create something like `context.getIdentity(identity: Identity | string)` for this, since it's being repeated quite a bit accross the entire project. If the passed value is an identity, we just return it. If it's a DID, we instantiate an identity, check if it `exists()`, return if so, throw if not.

Then, the `getIdentity` method in  the Polymesh class could use that behind the scenes